src/base/static/components/organisms/main-map.js

```diff
@@ -445,6 +451,21 @@ class MainMap extends Component {
         );
       }
     }
+
+    // Update place datasets when filters have been updated:
+    if (prevProps.filters.length !== this.props.filters.length) {
+      const datasetIds = this.props.layers
+        .filter(layerConfig => layerConfig.type === "place")
+        .map(layerConfig => layerConfig.id);
+      datasetIds.forEach(datasetId => {
+        const layerStatus = this.props.layerStatuses[datasetId];
+        const layerConfig = this.props.layers.find(
+          layer => layer.id === datasetId,
+        );
+        this._map.removeLayer(layerConfig, { clearCache: true });
+        this.addLayer(layerConfig, layerStatus.isBasemap);
```

Comment:
I'm wondering if we want to completely remove and re-add the layer on each filter application. There is another mapbox method, [`setData()`](https://www.mapbox.com/mapbox-gl-js/api/#geojsonsource#setdata), which allows you to change a source after it's been added to the map. Any layers built from that source are re-rendered.

I think this provides better overall performance, and should avoid the flicker that happens on the map now when you set and unset filters.

In `mapboxgl-provider.js` there is a method called `updateLayerData()` which I think we can use here. So we could replace the two lines above this comment with something like:

```
this._map.updateLayerData(
  layerConfig.id,
  createGeoJSONFromPlaces(
    this.props.places.filter(
      place => place.datasetId === layerConfig.id,
    ),
  ),
);
```

Reply:
Thanks for pointing this out. The UX, and the implementation, is much cleaner after this change :+1: